go/vt/vtgate/planbuilder/operators/update.go

```diff
@@ -399,8 +398,18 @@ func buildChildUpdOpForSetNull(ctx *plancontext.PlanningContext, fk vindexes.Chi
 			Right: compExpr,
 		}
 	}
+	// We only reach this code if foreign key checks are either unspecified or on.
```

Comment:
since this code is a copy-pasta from the `delete` operator, I suggest we extract it into a helper method and move the comment there

Reply:
Good idea. Fixed.
I had thought of doing that after I made all the changes, and then forgot.